CenterForOpenScience/osf.io
# [ENG-2159] Sets Cookie for Registration Detail Ember Page

## Purpose
There are a couple of bugs - one where the registration detail ember page was not provided with the cookie with a status message regarding the registration's approval/disproval and another where status messages set on cookies didn't have `id` fields leading to what looks like a translation error.

## Changes
* Adds `id` fields to status messages related to registrations (approval, embargo, public embargo, withdrawal)
* Sets a cookie with a status message for the RegistrationDetail page on the Ember app

## QA Notes
This can be partially tested through the UI. Follow the re-creation steps for the bug [here](https://openscience.atlassian.net/browse/ENG-2159) and you should see a registration approval message on the registration detail page after a moderator on the registration has clicked through the approval link. The message will still display mention of a translation issue because front-end changes are necessary to take advantage of the ids added in this PR. Note, this PR incorporates the backend changes for ENG-2160.

## Documentation
N/A

## Side Effects
N/A

## Ticket
[JIRA Ticket](https://openscience.atlassian.net/browse/ENG-2159)
